lib/classes/Variables.js

```diff
@@ -144,13 +144,17 @@ class Variables {
     return valueToPopulate;
   }
 
-  getValueFromFile(variableString) {
+  getValueFromFile(variableString, overwriteMode) {
     const matchedFileRefString = variableString.match(this.fileRefSyntax)[0];
     const referencedFileRelativePath = matchedFileRefString
       .replace(this.fileRefSyntax, (match, varName) => varName.trim());
     const referencedFileFullPath = path.join(this.serverless.config.servicePath,
       referencedFileRelativePath);
 
+    if (overwriteMode === true && !this.serverless.utils.fileExistsSync(referencedFileFullPath)) {
```

Comment:
same comment here, why not just check if the file doesn't exist return undefined?